lib/ThreeElementVectorLoweringPass.cpp

```diff
@@ -822,7 +794,7 @@ Value *clspv::ThreeElementVectorLoweringPass::visitStoreInst(StoreInst &I) {
   Value *EquivalentValue = visit(I.getValueOperand());
   Value *EquivalentPointer = nullptr;
 
-  if (I.getPointerOperand()->getType()->isOpaquePointerTy()) {
+  if (I.getPointerOperand()->getType()->isPointerTy()) {
```

Comment:
The if can be dropped since it's dealing with the pointer operand (it will always be a pointer).